src/utils/generateUID.js

```diff
@@ -0,0 +1,22 @@
+export const generateUID = (prefix, separator) => {
+  let counter = 1
+
+  let map = new WeakMap()
+
+  const get = item => {
+    if (!map.has(item)) {
+      map.set(item, counter++)
+      return get(item)
+    }
+    return `${prefix || 'uid'}${separator}${map.get(item)}`
```

Comment:
Would it be better if we use the array notation - like `rdts[0]`, `rdts[1]` and so on? I'm wondering if it'd help in a form submit. In that case, for a single instance, we would have to emit just `rdts`. We wouldn't know if there are multiple instances ahead of time so maybe this is a problem that the user needs to solve.

Reply:
I don't think it would be simple to detect single instances. For form submits we need to ensure the rootprefix/clientid is used as name for checkboxes as well if they should be grouped. I've passed the clientId down to the name in the latest commit.

Reply:
I don't think you're should use brackets in ids. Don't have any source for it now but I think Asp. NET MVC does some other naming when creating arrays of objects for submission. We can however append brackets on the name for checkboxes to group them in form submission.

Reply:
This is the reason https://www.w3.org/TR/html401/types.html#type-name
But that’s removed in html5 though.
Not sure how you would use a css selector for attributed on a id-tag with brackets though, even if an unlikely use case.

What about keeping the current format but adding brackets for the name (using clientid) of checkboxes to post them as an array? See example here:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#Using_checkbox_inputs

Then users could supply their own id, such as 'interests', in form submissions.

Reply:
Latest commits are up and running here: https://ellinge.github.io/react-dropdown-tree-select-test/

Reply:
I love the test page! If you don't mind, I think it should be included as a story on the docs site. 

Looking at the radio select, I think we should keep the dropdown open after clicking on a radio. This would help in accidental clicks in a long list and also would feel more in line with current behavior.

Reply:
About the test page, perhaps it's better with something like the Options-story? There’s a lot of dropdowns… The only reason I created it was to test it with IE11 properly. But I won’t mind if you want to use it.

About the radio select, what about an option at least to close on select or the other way around, to keep it open? We use the radio select with hidden inputs (similar to simple select) and as an tree in some cases for multi selects and are happy with its current state.

Reply:
> About the test page, perhaps it's better with something like the Options-story? There’s a lot of dropdowns… 

Yeah, I agree. We can think about it later. 

> About the radio select, what about an option at least to close on select or the other way around, to keep it open?

Sure, that's a good idea.